configure-x86.sh

```diff
@@ -0,0 +1,179 @@
+
+#---------------------------------------------------------------------
+# Description: This script is the *ONLY* place where "qemu*" build options
+# should be defined.
+#
+# Note to maintainers:
+#
+# XXX: Every option group *MUST* be documented explaining why it has
+# been specified.
+#---------------------------------------------------------------------
+
+typeset -a qemu_options
+multi_line=no
+
+#---------------------------------------------------------------------
+# Disabled options
+
+# bluetooth support not required
+qemu_options+=(--disable-bluez)
+
+# braille support not required
+qemu_options+=(--disable-brlapi)
+
+# Don't build documentation
+qemu_options+=(--disable-docs)
+
+# Disable GUI (graphics)
+qemu_options+=(--disable-curses)
+qemu_options+=(--disable-gtk)
+qemu_options+=(--disable-opengl)
+qemu_options+=(--disable-sdl)
+qemu_options+=(--disable-spice)
+qemu_options+=(--disable-vte)
+
+# Disable graphical network access
+qemu_options+=(--disable-vnc)
+qemu_options+=(--disable-vnc-jpeg)
+qemu_options+=(--disable-vnc-png)
+qemu_options+=(--disable-vnc-sasl)
+
+# Disable unused filesystem support
+qemu_options+=(--disable-fdt)
+qemu_options+=(--disable-glusterfs)
+qemu_options+=(--disable-libiscsi)
+qemu_options+=(--disable-libnfs)
+qemu_options+=(--disable-libssh2)
+qemu_options+=(--disable-rbd)
+
+# Disable unused compression support
+qemu_options+=(--disable-bzip2)
+qemu_options+=(--disable-lzo)
+qemu_options+=(--disable-snappy)
+
+# SECURITY: Disable unused security options
+qemu_options+=(--disable-seccomp)
+qemu_options+=(--disable-tpm)
+
+# Disable userspace network access ("-net user")
+qemu_options+=(--disable-slirp)
+
+# Disable USB
+qemu_options+=(--disable-libusb)
+qemu_options+=(--disable-usb-redir)
+
+# SECURITY: Don't build a static binary (lowers security)
+qemu_options+=(--disable-static)
+
+# Not required as "-uuid ..." is always passed to the qemu binary
+qemu_options+=(--disable-uuid)
+
+# Disable debug
+qemu_options+=(--disable-debug-tcg)
+qemu_options+=(--disable-qom-cast-debug)
+qemu_options+=(--disable-tcg-interpreter)
+qemu_options+=(--disable-tcmalloc)
+
+# SECURITY: Disallow network downloads
+qemu_options+=(--disable-curl)
+
+# Disable Remote Direct Memory Access (Live Migration)
+# https://wiki.qemu.org/index.php/Features/RDMALiveMigration
+qemu_options+=(--disable-rdma)
+
+# Don't build the qemu-io, qemu-nbd and qemu-image tools
+qemu_options+=(--disable-tools)
+
+# Disable XEN driver
+qemu_options+=(--disable-xen)
+
+# FIXME: why is this disabled?
+# (for reference, it's explicitly enabled in Ubuntu 17.10 and
+# implicitly enabled in Fedora 27).
+qemu_options+=(--disable-linux-aio)
+
+# In "passthrough" security mode
+# (-fsdev "...,security_model=passthrough,..."), qemu uses a helper
+# application called virtfs-proxy-helper(1) to make certain 9p
+# operations safer. We don't need that, so disable it (and it's
+# dependencies).
+qemu_options+=(--disable-virtfs)
```

Comment:
Does it disable 9pfs support completely?

Reply:
nop, 9p still works

Reply:
Looking at `hw/Makefile.objs` and `hw/9pfs/Makefile.objs`, I don't think 9pfs is still supported if we have `--disable-virtfs`.

Besides, `virtfs-proxy-helper` introduces extra process management and extra data copy. Please enable `virtfs` so that runv can still use `local` fstype.

Reply:
oks